### PR TITLE
Updated Deployment concept docs

### PR DESCRIPTION
**Pull request by liyinan926**
This PR updates the Deployment concept doc and the associated Deployment YAMLs. 

@janetkuo @kow3ns

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/6494)

### CODE REVIEW

**Comment by zacharysarah** [comment]
Docs LGTM :shipit: 

/assign @janetkuo @kow3ns for tech review

File: docs/concepts/workloads/controllers/deployment.md [hunk 1]
```diff
@@ -937,7 +937,7 @@ a Pod is considered ready, see [Container Probes](/docs/concepts/workloads/pods/
 
 ### Rollback To
 
-Field `.spec.rollbackTo` has been deprecated in API versions `extensions/v1beta1` and `apps/v1beta1`, and is no longer supported in API version `apps/v1beta2`. Instead, `kubectl rollout undo` as introduced in [Rolling Back to a Previous Revision](#rolling-back-to-a-previous-revision) should be used.
+Field `.spec.rollbackTo` has been deprecated in API versions prior to `apps/v1beta2`, and is no longer supported in API versions starting `apps/v1beta2`. Instead, `kubectl rollout undo` as introduced in [Rolling Back to a Previous Revision](#rolling-back-to-a-previous-revision) should be used.
```
Comment (kow3ns) [inline docs/concepts/workloads/controllers/deployment.md]: Might want to leave this as explicitly deprecated in extensions/v1beta1

Comment (liyinan926) [reply docs/concepts/workloads/controllers/deployment.md]: Done.

**Comment by janetkuo** [comment]
LGTM, please squash

**Pull request merged**